### DISCUSSION TITLE
Can we reduce the prompt size of entity_extraction.txt, it takes most of tokens consumption

### DISCUSSION THREAD

**Discussion by KylinMountain** [microsoft/graphrag#460] [General]
It is too large, it has 10835 words, about 2224 tokens.
I have txt with 26000 words, chunk it with 300 overlap 100, there's about 260 chunk to be processed with this entity_extraction.
It will consume about 260*(2224+300)=634240 tokens....

I think this is the most part of tokens consumption.

**Comment by AlonsoGuevara** [comment]
Hi @KylinMountain 
I have tested with smaller and bigger prompts. I think you can go with just one example on the extraction prompt, to reduce size, however, I haven't fully tested yet how much precision we lose in the event of less examples provided.

**Reply by KylinMountain** [reply]
@AlonsoGuevara 
Thanks for the reply, I think maybe we can try increase the default chunk size, which can also reduce tokens, but not know if this will reduce the precision...

Like 26000 words, chunk it with 1000 overlap 100, there will be only 26 times entity extraction. this will only consume about 83k tokens, it reduce about 550K tokens consumption.
What do you think?

**Comment by KylinMountain** [comment]
From my real test, it will take only 5-7 minutes which need take 30-60 min before.  And the result is better, it get more details information like someone's alias name is able to tell.
And the token consumption is reduce a lot.
Before:
![image](https://github.com/microsoft/graphrag/assets/3996229/59a57f75-834b-4525-b7d1-fa4e5aac33eb)
After: 
![image](https://github.com/microsoft/graphrag/assets/3996229/bfc28038-9e16-44ab-9c35-bb9e8ee90351)

**Reply by kShines** [reply]
Where can this chunk size be adjusted, please?

**Reply by rohit-metrum-ai** [reply]
you can adjust it in .yaml file .

**Reply by rohit-metrum-ai** [reply]
https://microsoft.github.io/graphrag/posts/config/json_yaml/

**Reply by KylinMountain** [reply]
> Where can this chunk size be adjusted, please?

@kShines it is located in settings.yaml, item name is chunk. If you pulled the latest main branch, the default chunk has been updated to 1200 and overlap is 100.
